pyvista/core/filters/poly_data.py

```diff
@@ -1671,22 +1694,56 @@ def decimate(
         >>> decimated = sphere.decimate(0.75)
         >>> decimated.plot(show_edges=True, line_width=2)
 
+        Decimate taking scalars attributes into account:
+
+        >>> sphere.decimate(0.5, scalars = True)
+
+        Decimate taking all attributes **except** normals into account:
+
+        >>> sphere.decimate(0.5, enable_all_attribute_error = True, normals = False)
+
         See :ref:`decimate_example` for more examples using this filter.
 
         """
+
         if not self.is_all_triangles:  # type: ignore[attr-defined]
             raise NotAllTrianglesError('Input mesh for decimation must be all triangles.')
 
+        has_attribute_error = kwargs.pop('attribute_error', False)
+        if has_attribute_error:  # pragma: no cover
+            warnings.warn(
+                "Since 0.45, use of `attribute_error=True` is deprecated."
+                "Use 'enable_all_attribute_error' instead.",
+                PyVistaDeprecationWarning,
+            )
+            enable_all_attribute_error = True
```

Comment:
For backwards compatibility maybe we should also set normals=False here
```suggestion
            enable_all_attribute_error = True
            if normals is None:
                normals = False
```